### PR TITLE
stats API

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

**Comment by dpo** [comment]
This is a follow-up to #25 after @abelsiqueira's comments.

cc @sshin23.

File: src/stats.jl [hunk 1]
```diff
@@ -105,20 +140,136 @@ function GenericExecutionStats(
...
+and mark it as reliable.
+"""
+function set_solver_specific!(stats::GenericExecutionStats, field::Symbol, value)
+  if field ∈ keys(stats.solver_specific)
+    try
+      # will fail if typeof(solver_specific[field]) does not support broadcast
+      stats.solver_specific[field] .= value
+    catch
+      stats.solver_specific[field] = value
+    end
```
Comment (abelsiqueira) [inline src/stats.jl:265]: Maybe you can prepend with `@.` here and it just works?

**Review by abelsiqueira — APPROVED** [review]
Thanks, I made a small suggestion, but it can be merged either way.

**Review by geoffroyleconte — APPROVED** [review]

File: src/stats.jl (hunk 1 above)
Comment (dpo) [reply src/stats.jl:265]: It doesn't unfortunately. Maybe because it's a Dict?

Comment (abelsiqueira) [reply src/stats.jl:265]: Probably it can't infer. We can use `if value isa AbstractArray` instead of try?

Comment (dpo) [reply src/stats.jl:265]: We don't know that they'll be arrays though. Custom types can support broadcasting.

**Review by tmigot — APPROVED** [review]
Looks good for me too. Thansk

File: src/stats.jl (hunk 1 above)
Comment (abelsiqueira) [reply src/stats.jl:265]: I don't think there is an easy way to find out. Let's leave it this way then.

**Pull request merged**